packages/ui-components/src/Input/Field.tsx

```diff
@@ -1,70 +1,38 @@
-import React, { FunctionComponent, useState } from "react";
+import React from "react";
 import { Field, FieldProps, FieldRenderProps } from "react-final-form";
-
-import {
-  SingleLineTextInput,
-  TextInputProps,
-  NumberInput,
-  NumberProps,
-} from "./TextTypeInput";
-import {
-  EmailInput,
-  NewPasswordInput,
-  ExistingPasswordInput,
-  EmailProps,
-  NewPasswordProps,
-  ExistingPasswordProps,
-  PasswordInputType,
-} from "./EmailPasswordInput";
-import { CheckboxInput, CheckboxInputProps } from "./CheckboxInput";
-import Icon from "../Icon/Icon";
-
-// the following omit statements can be removed
-// once type better limits what props can be passed in
-type TextInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    TextInputProps,
-  "type"
->;
-
-type NumberInputFieldProps = Omit<
-  FieldProps<number, FieldRenderProps<number, HTMLElement>, HTMLElement> &
-    NumberProps,
-  "type"
->;
-
-type CheckboxInputFieldProps = Omit<
-  FieldProps<boolean, FieldRenderProps<boolean, HTMLElement>, HTMLElement> &
-    CheckboxInputProps,
-  "type"
->;
-
-type EmailInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    EmailProps,
-  "type"
->;
-
-type NewPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    NewPasswordProps,
-  "type"
->;
-
-type ExistingPasswordInputFieldProps = Omit<
-  FieldProps<string, FieldRenderProps<string, HTMLElement>, HTMLElement> &
-    ExistingPasswordProps,
-  "type"
->;
-
-type InternalFieldProps =
-  | ({ InputFieldComponent: React.FunctionComponent } & TextInputFieldProps)
-  | CheckboxInputFieldProps
-  | EmailInputFieldProps
-  | NewPasswordInputFieldProps
-  | ExistingPasswordInputFieldProps;
-
-const InputField = ({
+import { SingleLineTextInput, NumberInput } from "./TextTypeInput";
+import { EmailInput, PasswordInput } from "./EmailPasswordInput";
+import { CheckboxInput } from "./CheckboxInput";
+
+// not including multiline input as it depends on Figma designs
+type InputTypes =
+  | typeof PasswordInput
+  | typeof EmailInput
+  | typeof SingleLineTextInput
+  | typeof NumberInput
+  | typeof CheckboxInput;
+
+// Extends field render props, which will be passed to the Field component in InputField
+export interface InputFieldProps<
+  IComponent extends React.ComponentType<InputTypes>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+> extends FieldProps<any, FieldRenderProps<any, HTMLElement>, HTMLElement> {
+  as: IComponent;
+  // should include types currently supported
+  type: "text" | "email" | "password" | "number" | "checkbox";
```

Comment:
Wanted to highlight this change, as it requires an additional prop to be passed whereas the field architecture prior (with one field per input - EmailField/PasswordField/TextInput etc) didn't require this. The additional prop is `type`: I've made `type` mandatory as well as the `as` prop, so when a developer wants to include a password field they'd do so `<InputField as={EmailInput} type="email" ...>`. Is this OK?

I believe this is necessary for the InputField to be able to support different Inputs. I tried doing some type checking with `as`, to see if we can pass the `type` prop ourselves depending on what `as` is, but I think it's not as safe as the developer passing it in themselves.